pkg/rc/replication_controller.go

```diff
@@ -201,18 +201,22 @@ func (rc *replicationController) addPods(current PodLocations) error {
 	// TODO: With Docker or runc we would not be constrained to running only once per node.
 	// So it may be the case that we need to make the Scheduler interface smarter and use it here.
 	possible := sets.NewString(eligible...).Difference(sets.NewString(currentNodes...))
+
+	// Users want deterministic ordering of nodes being populated to a new
+	// RC. Move nodes in sorted order by hostname to achieve this
+	possibleSorted := possible.List()
```

Comment:
Read the source https://github.com/kubernetes/kubernetes/blob/master/pkg/util/sets/string.go#L168.
Should I sort again on top of that in case that code changes?